app/services/notify_service.rb

```diff
@@ -177,6 +180,7 @@ def call(recipient, type, activity)
     return if dismiss?
 
     @notification.filtered = filter?
+    @notification.group_key = notification_group_key
```

Comment:
Are there scenarios where we want to create `Notification` records but NOT generate the group key? If yes, this is fine -- if no, could/should this be a callback on the model?

Reply:
I can't find of any such scenario. I did it in the service rather than the model for a few reasons, namely:
- it is the only place where we need to do it (it's not a strong reason to not do it in the model though)
- it was initially more complex (not that relevant anymore)
- it initially relied on the `id` of the created notification (that is not relevant anymore)
- it uses Redis in a way the model doesn't afaik

Those reasons have been greatly weakened since, so I'll consider moving it to the model.